src/analysis/seminorm.lean

```diff
@@ -0,0 +1,217 @@
+/-
+Copyright (c) 2019 Jean Lo. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jean Lo
+-/
+
+import algebra.pointwise
+import analysis.normed_space.basic
+
+/-!
+# Seminorms and local convexity
+
+This file introduces the following notions, defined for a vector space
+over a normed field:
+
+- the subset properties of being `absorbent` and `balanced`,
+
+- a `seminorm`, a function to the reals that is positive-semidefinite,
+  absolutely homogeneous, and subadditive.
+
+We prove related properties.
+
+(TODO:) define and show equivalence of two notions of local convexity
+for a t.v.s. over ℝ or ℂ: that it has a local base of balanced convex
+absorbent sets, and that it carries the initial topology induced by a
+family of seminorms,
+
+## References
+* [H. H. Schaefer, *Topological Vector Spaces*][schaefer1966]
+-/
+
+-- subset properties : absorbent and balanced sets in a vector space
+-- over a nondiscrete normed field
+section
+
+variables
+(𝕜 : Type*) [nondiscrete_normed_field 𝕜]
+{E : Type*} [add_comm_group E] [vector_space 𝕜 E]
+
+open set normed_field
+
+/-- a set `A` absorbs another set `B` if `B` is contained in scaling
```

Comment:
Here and below, please capitalize the first letter of complete sentences for consistency.

```suggestion
/-- A set `A` absorbs another set `B` if `B` is contained in scaling
```

Reply:
This and subsequent edits committed in 49aed8283d731ccbf27d378c47c9a19e82038a84 . Thanks for the catch! Resolving the comments now.